fvm/contractFunctionInvoker.go

```diff
@@ -41,17 +40,13 @@ func NewContractFunctionInvoker(
 	}
 }
 
-func (i *ContractFunctionInvoker) Invoke(env Environment, parentTraceSpan otelTrace.Span) (cadence.Value, error) {
+func (i *ContractFunctionInvoker) Invoke(envCtx *EnvContext, env Environment) (cadence.Value, error) {
 
-	ctx := env.Context()
-	if ctx.Tracer != nil && parentTraceSpan != nil {
-		span := ctx.Tracer.StartSpanFromParent(parentTraceSpan, trace.FVMInvokeContractFunction)
-		span.SetAttributes(i.logSpanAttrs...)
+	span := envCtx.StartSpanFromRoot(trace.FVMInvokeContractFunction)
```

Comment:
could `envCtx` possibly be `nil` here? is it necessary to check nil for `envCtx` here? same question for `env`